### PR TITLE
support getting core documents

### PR DESCRIPTION
**Pull request by charleenfei**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: #631

### CODE REVIEW

File: p2p/common/protocol.go [hunk 1]
```diff
@@ -61,6 +65,10 @@ func MessageTypeFromString(mt string) MessageType {
 		found = MessageTypeSendAnchoredDoc
 	} else if MessageTypeSendAnchoredDocRep.Equals(mt) {
 		found = MessageTypeSendAnchoredDocRep
+	} else if MessageTypeGetDoc.Equals(mt) {
+		found = MessageTypeGetDoc
+	} else if MessageTypeGetDocRep.Equals(mt) {
+		found = MessageTypeGetDocRep
```
Comment (lucasvo) [inline p2p/common/protocol.go]: @charleenfei could you change this entire block to use a switch statement instead?

Comment (charleenfei) [reply p2p/common/protocol.go]: will do!

File: p2p/receiver/handler.go [hunk 2]
```diff
@@ -190,6 +193,47 @@ func (srv *Handler) SendAnchoredDocument(ctx context.Context, docReq *p2ppb.Anch
...
+	p2pEnv, err := p2pcommon.PrepareP2PEnvelope(ctx, nc.GetNetworkID(), p2pcommon.MessageTypeGetDocRep, res)
+	if err != nil {
+		return convertToErrorEnvelop(err)
+	}
+
+	return p2pEnv, nil
+}
+
+// GetDocument receives document identifier and retrieves the corresponding CoreDocument from the repository
+func (srv *Handler) GetDocument(ctx context.Context, docReq *p2ppb.GetDocumentRequest) (*p2ppb.GetDocumentResponse, error) {
```
Comment (vedhavyas) [inline p2p/receiver/handler.go]: Let's move this to the above function itself and rename `HandleGetDocument` -> `GetDocuemnt` ?

Comment (charleenfei) [reply p2p/receiver/handler.go]: That logic would differ from how the other methods on the handlers work (SendAnchoredDoc, ReqDocumentSignature). I agree there is redundancy here, I have noted this in a separate issue #651. Do you think it makes sense to address that in a separate PR?

**Pull request merged**